### PR TITLE
Desktop: Skip invalid scroll positions when receiving CodeMirror scroll updates

### PR DESCRIPTION
**Pull request by roman-r-m**
Resolves https://github.com/laurent22/joplin/issues/4797

### CODE REVIEW

**Comment by tessus** [comment]
I'm not sure I understand. At least part of the fix or the root cause.

How can height == clientHeight, if the note is many pages long, but the height of the client is just a fraction of that?
If that were the case, it would also mean that the scroll position is not going to be set even though the position was in the middle of the note.
Maybe I'm missing something here. I haven't slept well, so it's very much possible that my brain is not working properly.

**Comment by roman-r-m** [comment]
> How can height == clientHeight, if the note is many pages long, but the height of the client is just a fraction of that?

Not sure tbh. I saw it with debugger. It seems to occur only when switching notes so might be some side effect.
Maybe @CalebJohn would know?

**Comment by CalebJohn** [comment]
Looks like it's a race condition when the editor is initialized (which happens whenever notes are switched). At some point before being fully initialized CodeMirror sets `height === clientHeight` and it's whats causing the NaN. 

There's a few weird things happening here with CodeMirror, for example `editor_scroll` is being called even though the user probably hasn't actually scrolled yet. 

If this does indeed improve the situation I would suggest we go ahead and merge it. 
I've been meaning to dig into the Joplin scroll behaviour at some point in order to add proper sync scroll, so will try to figure out the other oddities at that time.

**Comment by roman-r-m** [comment]
> If this does indeed improve the situation I would suggest we go ahead and merge it.

It fixed the problem in my tests. I have tried switching to another note (and back) via markdown link or simply by clicking on another note in the list.
In both cases Joplin restores more or less correct scroll position.

**Comment by tessus** [comment]
Using a test for percentage makes perfect sense. I'm sure it will help, I'm just not sure if it will fix all issues. It certainly won't hurt to merge this. On the contrary, making sure the position isn't set to some bogus value is an important precaution.

**Comment by tessus** [comment]
Laurent, can we merge this?

**Comment by tessus** [comment]
I've been running with this commit for a few days now. Unfortunately I still see random jumps when switching notes. There are no errors in the console or log though.
While I think it makes sense to merge this, it does not fix #4797 - or at least not completely.

**Comment by laurent22** [comment]
If the percent can sometimes be NaN, perhaps if we find the root cause it would make for a more reliable fix?

**Comment by CalebJohn** [comment]
@laurent22 I think the scroll behaviour deserves a bigger update and I plan on getting to that eventually. For now I recommend merging because it does improve the scroll positioning.

File: packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/index.ts [hunk 1]
```diff
@@ -70,8 +70,13 @@ export function useScrollHandler(editorRef: any, webviewRef: any, onScroll: Func
 
 		if (editorRef.current) {
 			const percent = editorRef.current.getScrollPercent();
-
-			setViewerPercentScroll(percent);
+			if (percent) {
```
Comment (laurent22) [inline packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/index.ts:73]: With this change, doesn't it mean that if you want to scroll to the top (percent = 0), you cannot anymore? Maybe we should check using `!isNaN(percent)` instead?

Comment (roman-r-m) [reply packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/index.ts]: Good point, thanks. Fixed

**Comment by laurent22** [comment]
All good now, thanks!

**Pull request merged**